edb/graphql/types.py

```diff
@@ -395,7 +400,7 @@ def get_fields(self, typename):
         if typename == 'Query':
             for name, gqltype in sorted(self._gql_interfaces.items(),
                                         key=lambda x: x[1].name):
-                if name in TOP_LEVEL_TYPES:
+                if name in TOP_LEVEL_TYPES or gqltype.name.startswith('_edb'):
```

Comment:
What's `_edb`?

Reply:
Basically it's an indicator that the underlying EdgeDB entity is some kind of internal thing (has a name starting with `__`, which is illegal for user-land). This is true of nested aliased types that are auto-generated, for example. It makes sense for them to exist as types in the GraphQL schema, but they should not be exposed at top-level queriable things.

See [the code](https://github.com/edgedb/edgedb/blob/15980391ce5535e938c45f836548efed1307c360/edb/graphql/types.py#L242) where this name translation happens.

I can add a comment here to clarify, though.

Reply:
> has a name starting with __, which is illegal for user-land

Yeah, that's why I asked, because I see only one `_` underscore, not two.

Reply:
I think that at some point we were debating whether to use `__` or `_` prefix in GraphQL-reflected names for internal stuff and decided that `__` was probably better left to the actual GraphQL-specific things (`__typename`, etc.), but our own translated internal entities would be prefixed with `_`, which later got to be `_edb` to potentially allow marking other flavours of internal stuff with a different `_something` prefix.

Reply:
Are the `_edb` things visible in introspection? I'd suggest to use `__edb_` prefix -- it's unlikely to clash with anything in GraphQL and is distinctly a private system name.

Reply:
I've added a comment regarding the choice of prefix and its meaning.